### PR TITLE
#3 Cherry pick fix rich text not work on android (#2475)

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2475

## Dependent

- Need merged: 

    - https://github.com/linagora/rich-text-composer/pull/28
    - https://github.com/linagora/html-editor-enhanced/pull/36

## Demo

[demo'.webm](https://github.com/linagora/tmail-flutter/assets/80730648/03e1d6f9-6e50-4690-864d-70719d25d0e9)

### CODE REVIEW

File: lib/features/manage_account/presentation/vacation/vacation_controller.dart [hunk 1]
```diff
@@ -92,6 +94,16 @@ class VacationController extends BaseController {
...
   }
 
+  void _initFocusListener() {
+    subjectTextFocusNode.addListener(() {
+      if (subjectTextFocusNode.hasFocus == true) {
+        if (PlatformInfo.isMobile) {
+          richTextControllerForMobile.hideRichTextView();
+        }
+      }
+    });
```
Comment (tddang-linagora) [inline lib/features/manage_account/presentation/vacation/vacation_controller.dart:104]: Create method to call `subjectTextFocusNode.removeListener` later in `onClose()`. Avoid using anonymous function inside `listener`.

Comment (dab246) [reply lib/features/manage_account/presentation/vacation/vacation_controller.dart]: done

**Review by tddang-linagora — APPROVED** [review]

**Comment by dab246** [comment]
Rebase to the `refactor` branch is complete

**Comment by hoangdat** [comment]
- [ ] still have the problem of focus when we change focus from `recipient fields` to `body editor`

https://github.com/linagora/tmail-flutter/assets/6462404/ca343fb0-3871-42c6-b96b-b87866f154f4

**Comment by dab246** [comment]
> * [ ]  still have the problem of focus when we change focus from `recipient fields` to `body editor`
> 
>  Screenrecorder-2024-06-25-09-35-55-238.mp4

What is the problem here?

**Comment by hoangdat** [comment]
> What is the problem here?

when we change focus to body editor, we still see that keyboard up and down, especially, sometimes down and never up again. How to prevent keyboard up and down when we move focus from recipient fields to editor?

**Comment by hoangdat** [comment]
- [ ] sometime toolbar is not showed exactly selected option (please check video below from 1:20)

**Comment by dab246** [comment]
> * [ ]  sometime toolbar is not showed exactly selected option (please check video below from 1:20)
> 
>  Screenrecorder-2024-06-25-09-49-40-272.mp4

Can't see exactly what the error is here. Please rewind the video for a few seconds to see the specific error.

**Comment by hoangdat** [comment]
> Can't see exactly what the error is here.

Please play video around 1:13, we also have `B`, `I`, `U`, `S` in editor, but toolbar only show `I`, `U`, `S`

**Comment by tddang-linagora** [comment]
Tested on Android

- [ ] Bold
- [ ] Italic
- [ ] Strikethrough
- [ ] Underlined
- [ ] Text color
- [ ] Background color

**Comment by hoangdat** [comment]
- [x] Chrome

File: lib/features/manage_account/presentation/menu/settings/settings_view.dart [hunk 2]
```diff
@@ -217,7 +242,9 @@ class SettingsView extends GetWidget<SettingsController> {
           }
         case AccountMenuItem.forward:
           if (controller.manageAccountDashboardController.isForwardCapabilitySupported) {
-            return ForwardView();
+            return SafeArea(
+              top: false,
+              child: ForwardView());
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/menu/settings/settings_view.dart:247]: why we need to add `SafeArea` like that?

Comment (dab246) [reply lib/features/manage_account/presentation/menu/settings/settings_view.dart:247]: We need horizontal padding for landscape mobile.
 
<table>
 <tr>
    <th>Before</th>
    <th>After</th>
  </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>

**Comment by dab246** [comment]
> * [ ]  still have the problem of focus when we change focus from `recipient fields` to `body editor`
>   when we change focus to `body editor`, we still see that keyboard up and down, especially, sometimes down and never up again. How to prevent keyboard up and down when we move focus from `recipient fields` to `editor`?
> 
>  Screenrecorder-2024-06-25-09-35-55-238.mp4

Only happens on Android. iOS works fine. On Android, if you persist the soft keyboard, the caret cursor will be duplicated in the WebView and TextField. Therefore, as a workaround, we can only reopen the soft keyboard when moving focus from TextField to WebView

**Comment by dab246** [comment]
> * [ ]  sometime toolbar is not showed exactly selected option (please check video below from 1:20)
> 
>  Screenrecorder-2024-06-25-09-49-40-272.mp4

Fixed.

**Comment by tddang-linagora** [comment]
Tested on iOS

- [x] Bold
- [x] Italic
- [x] Strikethrough
- [x] Underlined
- [x] Text color
- [x] Background color

**Comment by hoangdat** [comment]
- Compromised solution: to prevent double cursor, take the disadvantage of blink blink of keyboard

**Comment by hoangdat** [comment]
> Fixed

Still have problem with `Bold` in my Android

**Comment by tddang-linagora** [comment]
Text format randomly turned off when typing text on start of new line

Ipad air 5th gen - debug mode - preprod env

https://github.com/linagora/tmail-flutter/assets/160106668/a005bf98-8eed-4315-8438-b0a5b78985a4

**Comment by dab246** [comment]
> > Fixed
> 
> Still have problem with `Bold` in my Android

Fixed. Please test again. 


[demo.webm](https://github.com/linagora/tmail-flutter/assets/80730648/34b42d7c-a877-492a-90df-5c3ac36a60b3)

**Comment by dab246** [comment]
> Text format randomly turned off when typing text on start of new line
> 
> Ipad air 5th gen - debug mode - preprod env
> 
>  Screen.Recording.2024-06-28.at.15.15.33.mov

## Reproduce

- Select `underline` style to enable it
- Select `strikeThrough` style to enable it
- Enter some text. Now `underline` & `strikeThrough` are applied to text
- Select again `underline` style to disable it
- Enter some text. Now `underline` & `strikeThrough` do not apply to text

=> Turn on both (`unlineline` and `strikeThrough`). When you turn off one of them, both will turn off.

## Analyze root cause

- This error only occurs on `safari webkit` (iOS, Desktop, Browser). It's related to `underline` and `strikeThrough` css style incompatibility on webkit. With safari requires adding `-webkit-text-decoration`.

Screenshot 2024-06-29 at 16 32 20

- Some other mail clients also have similar problems as `GMail` or `Apple Mail`


https://github.com/linagora/tmail-flutter/assets/80730648/afc68674-622a-469e-b316-782a28e32e35


https://github.com/linagora/tmail-flutter/assets/80730648/2218bdd1-8f39-429d-91d2-14891cac75be



- Currently there is no solution offered to solve this problem.

## Conclusion

- Android (Mobile/Tablet): Work fine

[demo-android.webm](https://github.com/linagora/tmail-flutter/assets/80730648/79455323-f6e7-4e08-9499-f02d9bac3e62)

- IOS (Mobile/iPad): Not working

**Pull request merged**